codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +401,188 @@ public TextHeaders clear() {
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.add(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.add(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object value) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(value));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence commaSeparated(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0 ; i < end; i++) {
+                    singleValue.append(values[i]).append(", ");
+                }
+                singleValue.append(values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparated(Iterable<?> values) {
+            StringBuilder singleValue = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            while (iterator.hasNext()) {
+                singleValue.append(iterator.next());
+                if (iterator.hasNext()) {
```

Comment:
Consider a bit of loop unrolling to removing the duplicate `hasNext()` check in the general case:

```java
        StringBuilder singleValue = new StringBuilder();
        Iterator<?> iterator = values.iterator();
        Object next = iterator.hasNext() ? iterator.next() : null;
        if (next != null) {
            for (Object nextNext = iterator.hasNext() ? iterator.next() : null;
                    nextNext != null;
                    next = nextNext, nextNext = iterator.hasNext() ? iterator.next() : null) {
                singleValue.append(next).append(", ");
            }
            singleValue.append(next);
        }

        return singleValue.toString();
```

It may be a bit more convoluted but I'm interested in your thoughts.